ads/pixels.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {validateData, writeScript} from '../3p/3p';
+
+export function pixels(global, data) {
+  validateData(data,
+      ['origin', 'sid', 'tag'],
+      ['clickTracker', 'viewability']
+  );
+  data.tag = (data.tag).toLowerCase();
```

Comment:
this is up to you but any attributes could potentially come through as a number

Reply:
Updated.